src/coreclr/jit/importercalls.cpp

```diff
@@ -2261,6 +2256,26 @@ void Compiler::impPopArgsForSwiftCall(GenTreeCall* call, CORINFO_SIG_INFO* sig,
         arg = insertAfter->GetNext();
     }
 
+    assert(call->gtArgs.CountArgs() == sig->numArgs);
+
+    if (swiftErrorIndex != sig->numArgs)
+    {
+        // Before calling a Swift method that may throw, the error register must be cleared,
+        // as we will check for a nonzero error value after the call returns.
+        // By adding a well-known "sentinel" argument that uses the error register,
+        // the JIT will emit code for clearing the error register before the call,
+        // and will mark the error register as busy so it isn't used to hold the function call's address.
+        CallArg* const swiftErrorArg          = call->gtArgs.GetArgByIndex(swiftErrorIndex);
```

Comment:
The loop removes/inserts arguments, so this part still has to happen before the loop above.